### PR TITLE
Prep for 2.13.0 release

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by PikaJoyce — APPROVED** [review]
🚢  it!

File: lib/media/track/localvideotrack.js [hunk 1]
```diff
@@ -62,37 +62,151 @@ class LocalVideoTrack extends LocalMediaVideoTrack {
...
+   * class GrayScaleProcessor {
+   *   constructor() {
+   *     this.outputFrame = new OffscreenCanvas(0, 0);
+   *   }
+   *   processFrame(inputFrame) {
+   *     this.outputFrame.width = inputFrame.width;
+   *     this.outputFrame.height = inputFrame.height;
+   *
+   *     const context = this.outputFrame.getContext('2d');
+   *     context.filter = 'grayscale(100%)';
```
Comment (manjeshbhargav) [inline lib/media/track/localvideotrack.js:109]: I think this can go in the constructor.

File: lib/media/track/localvideotrack.js [hunk 2]
```diff
@@ -62,37 +62,151 @@ class LocalVideoTrack extends LocalMediaVideoTrack {
...
+      .catch(error => this._log.warn(
+        'setMediaStreamTrack failed on LocalVideoTrack RTCRtpSender', { error, mediaStreamTrack }))
+      .then(() => {
+        this._unprocessedTrack = useProcessed ? unprocessedTrack : null;
+      });
+  }
+
+  /**
+   * Add a {@link VideoProcessor} to allow for custom processing of video frames belonging to a VideoTrack.
+   * Only Chrome supports this as of now.
```
Comment (manjeshbhargav) [inline lib/media/track/localvideotrack.js:96]: Please mention about the exception raised when called on other browsers.

File: lib/media/track/remotevideotrack.js [hunk 3]
```diff
@@ -35,6 +35,78 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
...
+   * class GrayScaleProcessor {
+   *   constructor() {
+   *     this.outputFrame = new OffscreenCanvas(0, 0);
+   *   }
+   *   processFrame(inputFrame) {
+   *     this.outputFrame.width = inputFrame.width;
+   *     this.outputFrame.height = inputFrame.height;
+   *
+   *     const context = this.outputFrame.getContext('2d');
+   *     context.filter = 'grayscale(100%)';
```
Comment (manjeshbhargav) [inline lib/media/track/remotevideotrack.js:55]: This can go in the constructor.

File: lib/media/track/remotevideotrack.js [hunk 4]
```diff
@@ -35,6 +35,78 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
     super(sid, mediaTrackReceiver, isEnabled, isSwitchedOff, setPriority, options);
   }
 
+  /**
+   * Add a {@link VideoProcessor} to allow for custom processing of video frames belonging to a VideoTrack.
+   * When a Participant unpublishes and re-publishes a VideoTrack, a new RemoteVideoTrack is created and
+   * any VideoProcessors attached to the previous RemoteVideoTrack would have to be re-added again.
+   * Only Chrome supports this as of now.
```
Comment (manjeshbhargav) [inline lib/media/track/remotevideotrack.js:42]: Mention the exception raised when called in other browsers.

**Review by makarandp0 — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,56 @@ The Twilio Programmable Video SDKs use [Semantic Versioning](http://www.semver.o
...
+
+  ```ts
+  abstract class VideoProcessor {
+    abstract processFrame(inputFrame: OffscreenCanvas)
+      : Promise<OffscreenCanvas | null>
+      | OffscreenCanvas | null;
+  }
+  ```
+
+  A VideoTrack provides new methods `addProcessor` and `removeProcessor` which can be used to add and remove a VideoProcessor. It also provides a new property `processor` which points to the current VideoProcessor being used by the VideoTrack. For example, you can toggle a blur filter on a LocalVideoTrack as shown below.
```
Comment (manjeshbhargav) [inline CHANGELOG.md:22]:
```suggestion
  A VideoTrack provides new methods [addProcessor](url_to_localvideotrack_addprocessor_mdn_link) and [removeProcessor](url_to_localvideotrack_removeprocessor_mdn_link) which can be used to add and remove a VideoProcessor. It also provides a new property `processor` which points to the current VideoProcessor being used by the VideoTrack. For example, you can toggle a blur filter on a LocalVideoTrack as shown below.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,56 @@ The Twilio Programmable Video SDKs use [Semantic Versioning](http://www.semver.o
...
+    width: 1280,
+    height: 720
+  }).then(track => {
+    const processor = new BlurVideoProcessor(1280, 720, 5);
+    document.getElementById('preview').appendChild(track.attach());
+    document.getElementById('toggle-blur').onclick = () => track.processor
+      ? track.removeProcessor(processor)
+      : track.addProcessor(processor);
+  });
+  ```
```
Comment (manjeshbhargav) [inline CHANGELOG.md:53]: Please add an example for RemoteVideoTrack:
```js
room.on('trackSubscribed', track => {
  if (track.kind === 'video') {
    /* Set up processor for RemoteVideoTrack */
  }
});
```

**Review by manjeshbhargav — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,69 @@ The Twilio Programmable Video SDKs use [Semantic Versioning](http://www.semver.o
...
+    height: 720
+  }).then(track => {
+    const processor = new BlurVideoProcessor(1280, 720, 5);
+    document.getElementById('preview').appendChild(track.attach());
+    document.getElementById('toggle-blur').onclick = () => track.processor
+      ? track.removeProcessor(processor)
+      : track.addProcessor(processor);
+  });
+
+  // Remote video track
```
Comment (manjeshbhargav) [inline CHANGELOG.md:55]:
```suggestion
  \`\`\`
  
  You can also toggle a blur filter on a RemoteVideoTrack as shown below.
  
  \`\`\`js
```

**Review by manjeshbhargav — APPROVED** [review]

**Pull request merged**